### PR TITLE
create complaint in backend and router

### PR DESCRIPTION
**Pull request by Mai-Elhajeen**

### CODE REVIEW

File: client/src/router.tsx [hunk 1]
```diff
@@ -20,11 +22,13 @@ const router = createBrowserRouter([
...
     element: <AdminDashboard />,
     children: [
-      { index: true, element: <h1>statistics</h1> },
+      { index: true, element: <h1>لوحة التحكم</h1> },
       { path: 'services', element: <ServicesContainer /> },
       { path: 'services/addService', element: <AddService /> },
       { path: 'services/editService/:id', element: <EditService /> },
       { path: 'bills', element: <Bills /> },
+      { path: 'complaints', element: <Complaints /> },
+      { path: 'complaints/show/:id', element: <SingleComplaints /> },
```
Comment (Mu7ammadAbed) [inline client/src/router.tsx:31]: add single complaint as a children to complaints

File: client/src/Interfaces/complaints.ts [hunk 2]
```diff
@@ -0,0 +1,7 @@
+export interface InferComplaintsModel {
```
Comment (Mu7ammadAbed) [inline client/src/Interfaces/complaints.ts:1]: Edit this

**Review by Mu7ammadAbed — COMMENTED** [review]
Add tests to your routes

File: client/src/router.tsx [hunk 3]
```diff
@@ -31,11 +34,25 @@ const router = createBrowserRouter([
...
+      {
+        path: 'complaints',
+        element: <Complaints />,
+        children: [
+          {
+            index: true,
+            element: <DataTable />,
+          },
+          {
+            path: 'show/:id',
```
Comment (Mu7ammadAbed) [inline client/src/router.tsx:51]:
```suggestion
            path: ':id',
```

**Review by Mu7ammadAbed — APPROVED** [review]

**Review by mohjaps — APPROVED** [review]

**Review by Mu7ammadAbed — APPROVED** [review]

**Review by bayan2002 — APPROVED** [review]

**Pull request merged**